src/Controller/SubjectController.php

```diff
@@ -2,42 +2,106 @@
 
 namespace App\Controller;
 
-use App\Entity\Basket;
 use App\Entity\Subject;
+use App\Form\SubjectFormType;
 use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
 class SubjectController extends AbstractController
 {
+    private EntityManagerInterface $entityManager;
 
-    // #[Route('/subject', name: 'app_subject')]
-    // public function subject(EntityManagerInterface $entityManager): Response
-    // {
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
 
-    //     $user = $this->getUser();
-    //     if (!$user) {
-    //         $this->addFlash('error', 'Vous devez être connecté pour accéder au forum.');
-    //         return $this->redirectToRoute('app_login'); 
-    //     }
+    #[Route('/subject', name: 'app_subject')]
+    public function subject(Request $request): Response
+    {
+        $subjects = $this->entityManager->getRepository(Subject::class)->findAll();
+        $subject = $this->entityManager->getRepository(Subject::class)->findAll();
 
-    //     $subjectRepository = $entityManager->getRepository(Subject::class);
-    //     $existingSubjects = $subjectRepository->findAll();
+        return $this->render('subject/index.html.twig', [
+            'subjects' => $subjects,
+            'subject' => $subject,
+        ]);
+    }
 
-    //     if (!empty($existingSubjects)) {
-    //         return $this->render('subject/index.html.twig', [
-    //             'subjects' => $existingSubjects,
-    //         ]);
-    //     }
+    #[Route('/subject/create', name: 'app_subject_create')]
+    public function addSubject(Request $request): Response
+    {
+        $this->denyAccessUnlessGranted('ROLE_USER');
 
-    //     $subject = new Subject();
-    //     $subject->setName('Movies');
-    //     $subject->setDescription('Œuvre cinématographique enregistrée sur film (cinéma).');
+        $subject = new Subject();
+        $form = $this->createForm(SubjectFormType::class, $subject);
+        $form->handleRequest($request);
 
-    //     $entityManager->persist($subject);
-    //     $entityManager->flush();
+        if ($form->isSubmitted() && $form->isValid()) {
+            $subject->setCategory($form->get('theme')->getData());
```

Comment:
Pas besoin de faire un setCategory, si tu l'as selectionne ca se fait automatiquement.

Reply:
An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'category_id' cannot be null
J'ai cette erreur là quand j'enlève cette ligne